File: MusicEventManagementSystem/MusicEventManagementSystem.Core/Models/Entities/EventOrganization/Performer.cs
using MusicEventManagementSystem.Core.Models.Entities.PerformerCommunication;

namespace MusicEventManagementSystem.Core.Models.Entities.EventOrganization
{
    public class Performer
    {
        public int PerformerId { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string? Contact { get; set; }
        public string Genre { get; set; } = string.Empty;
        public int Popularity { get; set; }
        public string TechnicalRequirements { get; set; } = string.Empty;
        public decimal MinPrice { get; set; }
        public decimal MaxPrice { get; set; }
        public TimeSpan AverageResponseTime { get; set; }
        public string Status { get; set; } = string.Empty;
        public DateTime UpdatedAt { get; set; }

        // Navigation Properties
        public Negotiation? Negotiation { get; set; } // One-to-One relationship
        public ICollection<Contract> Contracts { get; set; } = new List<Contract>();
    }

}

File: MusicEventManagementSystem/MusicEventManagementSystem.Client/src/frontend/event-organization/types/form/performer.ts
export interface PerformerCreateForm {
  name: string;
  email: string;
  contact: string;
  genre: string;
  popularity: number;
  technicalRequirements: string;
  minPrice: number;
  maxPrice: number;
  averageResponseTime: string; // TimeSpan kao string (format: HH:mm:ss)
  status: string;
}

export interface PerformerUpdateForm {
  name?: string;
  email?: string;
  contact?: string;
  genre?: string;
  popularity?: number;
  technicalRequirements?: string;
  minPrice?: number;
  maxPrice?: number;
  averageResponseTime?: string;
  status?: string;
}